packages/workflows/src/definition/product/create-product-variants.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  TransactionStepsDefinition,
+  WorkflowManager,
+} from "@medusajs/orchestration"
+import { InputAlias, Workflows } from "../../definitions"
+import { exportWorkflow, pipe } from "../../helper"
+
+import { ProductTypes, WorkflowTypes } from "@medusajs/types"
+import { ProductHandlers } from "../../handlers"
+
+export enum CreateProductVariantsActions {
+  prepare = "prepare",
+  createProductVariants = "createProductVariants",
+  revertProductVariantsCreate = "revertProductVariantsCreate",
+  upsertPrices = "upsertPrices",
+}
+
+export const workflowSteps: TransactionStepsDefinition = {
+  next: {
+    action: CreateProductVariantsActions.prepare,
+    noCompensation: true,
+    next: {
+      action: CreateProductVariantsActions.createProductVariants,
+      noCompensation: true,
+      next: [
+        {
+          action: CreateProductVariantsActions.upsertPrices,
+        },
+      ],
+    },
+  },
+}
+
+const handlers = new Map([
+  [
+    CreateProductVariantsActions.prepare,
+    {
+      invoke: pipe(
+        {
+          merge: true,
+          inputAlias: InputAlias.ProductVariantsCreateInputData,
+          invoke: {
+            from: InputAlias.ProductVariantsCreateInputData,
+          },
+        },
+        ProductHandlers.createProductVariantsPrepareData
+      ),
+    },
+  ],
+  [
+    CreateProductVariantsActions.createProductVariants,
+    {
+      invoke: pipe(
+        {
+          merge: true,
+          invoke: {
+            from: CreateProductVariantsActions.prepare,
+          },
+        },
+        ProductHandlers.createProductVariants
+      ),
+    },
+  ],
+  [
+    CreateProductVariantsActions.upsertPrices,
+    {
+      invoke: pipe(
+        {
+          merge: true,
+          invoke: [
+            {
+              from: CreateProductVariantsActions.createProductVariants,
+            },
+          ],
+        },
+        ProductHandlers.upsertVariantPrices
+      ),
+      compensate: pipe(
+        {
+          merge: true,
+          invoke: [
+            {
+              from: CreateProductVariantsActions.prepare,
+            },
+            {
+              from: CreateProductVariantsActions.upsertPrices,
+            },
+          ],
+        },
+        ProductHandlers.revertVariantPrices
```

Comment:
todo: not directly related to this PR but I noticed when we revert price update in this handler we remove (remote) links and reupdate the original prices but we do not set links again for the original records

Reply:
Links shouldn't get updated. They remain consistent. I've now pushed the link creation only when its needed. Good catch!